controllers/snapshot/snapshot_adapter.go

```diff
@@ -466,7 +493,7 @@ func (a *Adapter) createIntegrationPipelineRun(application *applicationapiv1alph
 // If it's not possible to create it and set the application as the owner, an error will be returned
 func (a *Adapter) createSnapshotEnvironmentBindingForSnapshot(application *applicationapiv1alpha1.Application,
 	environment *applicationapiv1alpha1.Environment, snapshot *applicationapiv1alpha1.Snapshot,
-	components *[]applicationapiv1alpha1.Component) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
+	components *[]applicationapiv1alpha1.Component, optionalLabelKeyAndName ...string) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
```

Comment:
Wdyt about having the labels parameter as a `map[string]string{}` and putting the labels in that way?
That would remove the need for the [checks below](https://github.com/redhat-appstudio/integration-service/pull/218/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R509-R510) and would simply add what's in that map to the labels.

Reply:
Yes, let me update it.

Reply:
Thanks for the suggestions. Learned them.